clients/src/main/java/org/apache/kafka/common/protocol/ReadableBuf.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.protocol;
+
+/**
+ * A readable backed by some buffers
+ */
+public interface ReadableBuf extends Readable {
```

Comment:
Why not make this part of Readable?

Reply:
Thought about that, `Readable` only has methods related to reading byte/int/long/.... etc

It has no method to set things (like position) and also that feels more like a Buffer abstraction. Only. method that recently got added to it was `remaining` which I would argue should be added the way this PR has, create an abstraction for a Readable that is backed by a buffer.